src/lib/bootloader/stage1_device.rb

```diff
@@ -53,61 +56,66 @@ def underlaying_devices_for(dev)
       res.uniq
     end
 
+    def pv_disk(vg)
+      pvs = usable_pvs(vg)
+      pvs.map { |p| p.ancestors.find { |a| a.is?(:disk) }.name }
+    end
+
     # get one level of underlaying devices, so no recursion deeper
     def underlaying_devices_one_level(dev)
-      tm = Yast::Storage.GetTargetMap
-      disk_data = Yast::Storage.GetDiskPartition(dev)
-      if disk?(disk_data)
-        disk = Yast::Storage.GetDisk(tm, dev)
-        # md disk is just virtual device, so select underlaying device /boot partition
-        case disk["type"]
-        when :CT_MD then return underlaying_disk_with_boot_partition
-        when :CT_LVM
-          res = lvm_underlaying_devices(disk)
-          return res.map { |r| Yast::Storage.GetDiskPartition(r)["disk"] }
-        when :CT_DMRAID
-          return disk["devices"]
-        end
-      # given device is partition
-      else
-        part = Yast::Storage.GetPartition(tm, dev)
-        if part["type"] == :lvm
-          lvm_group = Yast::Storage.GetDisk(tm, disk_data["disk"])
-          return lvm_underlaying_devices(lvm_group)
-        elsif part["type"] == :sw_raid
-          return devices_on(part)
-        elsif part["fstype"] == Yast::Partitions.dmraid_name
-          mapper = Yast::Storage.GetDisk(tm, disk_data["disk"])
-          return mapper["devices"]
-        end
+      # FIXME: there is nothing like this right now in libstorage-ng
+      #  a_vg.name #=> "/dev/vgname"
+      # revisit when such thing exist
+      match = /^\/dev\/(\w+)$/.match(dev)
+      if match && match[1]
+        vg = devicegraph.lvm_vgs.find { |v| v.vg_name == match[1] }
+        return pv_disk(vg) if vg
       end
 
+      lvs = devicegraph.lvm_lvs.find { |v| v.name == dev }
+      return usable_pvs(lvs.lvm_vg).map { |v| v.blk_device.name } if lvs && lvs.lvm_vg
+
+      # TODO: storage-ng
+      # md
+      # md disk is just virtual device, so select underlaying device /boot partition
+      # return underlaying_disk_with_boot_partition
+
+      # TODO: storage-ng
+      # DMRAID main device - not implemented in libstorage-ng
+      # return disk["devices"]
+
+      # TODO: storage-ng
+      # DMRAID part device - not implemented in libstorage-ng
+      # return mapper["devices"]
+      #
+      # TODO: storage-ng
+      # sw_raid
+      # return devices_on(part)
```

Comment:
well, it is already implemented in yast2-storage, so maybe it make sense to implement it?